### PR TITLE
Add dev.rdm.RDM

### PR DESCRIPTION
**Pull request by uglide**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. 
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: dev.rdm.RDM.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "command": "rdm",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--share=network",
+        "--filesystem=home",
```
Comment (hfiguiere) [inline dev.rdm.RDM.json:12]: does it really need to access the whole home directory?

File: dev.rdm.RDM.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            "post-install": [
+                "install -Dm644 icon.svg /app/share/icons/hicolor/scalable/apps/dev.rdm.RDM.svg",
+                "install -Dm644 dev.rdm.RDM.metainfo.xml -t /app/share/metainfo",
+                "install -Dm644 dev.rdm.RDM.desktop -t /app/share/applications"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/uglide/RedisDesktopManager",
+                    "commit": "0b2a808f32e18421f64aae30221eb97caa838f06"
```
Comment (hfiguiere) [inline dev.rdm.RDM.json:35]: why isn't this the commit for `2021.7`

File: dev.rdm.RDM.desktop [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
```
Comment (barthalion) [inline dev.rdm.RDM.desktop:1]: Consider making it part of your code repository as it's not specific to Flathub.

File: dev.rdm.RDM.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline dev.rdm.RDM.metainfo.xml:1]: Consider making it part of your code repository as it's not specific to Flathub.

File: icon.svg [hunk 5]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (barthalion) [inline icon.svg:1]: Consider making it part of your code repository as it's not specific to Flathub.

File: dev.rdm.RDM.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
+{
+    "app-id": "dev.rdm.RDM",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "command": "rdm",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
```
Comment (barthalion) [inline dev.rdm.RDM.json:9]:
```suggestion
        "--socket=fallback-x11",
```

**Comment by barthalion** [comment]
bot, build dev.rdm.RDM

File: dev.rdm.RDM.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            ],
+            "post-install": [
+                "install -Dm644 icon.svg /app/share/icons/hicolor/scalable/apps/dev.rdm.RDM.svg",
+                "install -Dm644 dev.rdm.RDM.metainfo.xml -t /app/share/metainfo",
+                "install -Dm644 dev.rdm.RDM.desktop -t /app/share/applications"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/uglide/RedisDesktopManager",
```
Comment (barthalion) [inline dev.rdm.RDM.json:34]: You probably want to have a look at https://github.com/flathub/flatpak-external-data-checker/ to automate updates.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build dev.rdm.RDM

**Comment by hfiguiere** [comment]
it is missing some python dependencies.

File: dev.rdm.RDM.json (hunk 7 above)
Comment (uglide) [reply dev.rdm.RDM.json:34]: Thanks!

File: dev.rdm.RDM.json (hunk 1 above)
Comment (uglide) [reply dev.rdm.RDM.json:12]: RDM needs access to HOME dir to be able to import/export settings, ssh keys, etc.

Comment (hfiguiere) [reply dev.rdm.RDM.json:12]: where are the settings?

also can't is use the ssh agent instead?

**Comment by hfiguiere** [comment]
bot, build dev.rdm.RDM

File: dev.rdm.RDM.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            "post-install": [
+                "install -Dm644 ./src/resources/images/logo.svg /app/share/icons/hicolor/scalable/apps/dev.rdm.RDM.svg",
+                "install -Dm644 ./src/resources/flatpak/dev.rdm.RDM.metainfo.xml -t /app/share/metainfo",
+                "install -Dm644 ./src/resources/flatpak/dev.rdm.RDM.desktop -t /app/share/applications"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/uglide/RedisDesktopManager",
+                    "commit": "fe952001934598cc09ddac3817612adc7850f924",
```
[the remark this answers is not in the document]
Comment (uglide) [reply dev.rdm.RDM.json:35]: I use different commit because 2021.7 doesn't include flatpak specific files and changes

**Comment by uglide** [comment]
@hfiguiere What kind of distro/image does flathubbot is based on? Or how I can reproduce the issue with Python dependencies locally?

File: dev.rdm.RDM.json (hunk 1 above)
Comment (uglide) [reply dev.rdm.RDM.json:12]: Users may have older RDM versions installed (settings by default stored in $HOME/.rdm) and RDM needs access to it to provide a seamless migration to Flatpak variant of RDM.

**Comment by ariasmn** [comment]
@uglide Take a look at https://github.com/flatpak/flatpak-builder-tools/issues/208#issuecomment-875263449

I think that's the only thing missing (I used that flag in my local `pip-depencies.json`), to compile  **_pandas_** without errors. Don't know if a better approach is possible, maybe @hfiguiere can guide us.

**Comment by hfiguiere** [comment]
> @hfiguiere What kind of distro/image does flathubbot is based on? Or how I can reproduce the issue with Python dependencies locally?

By using `flatpak-builder`. I usually expect that the submission as be run that way.

File: dev.rdm.RDM.json (hunk 1 above)
Comment (hfiguiere) [reply dev.rdm.RDM.json:12]: `--filesystem=~/.rdm`

**Comment by uglide** [comment]
@hfiguiere locally it builds just fine with `flatpak-builder` so there are obviously some difference in build environments. That's why I'm asked for more details about flathub bot environment/os image

**Comment by barthalion** [comment]
I think the most important switch to pass to flatpak-builder is ` --sandbox`. The failure is likely caused by the fact Flathub builds are offline (which `--sandbox` does).

**Comment by uglide** [comment]
bot, build dev.rdm.RDM

**Comment by uglide** [comment]
bot, build dev.rdm.RDM

**Review by barthalion — APPROVED** [review]

**Comment by uglide** [comment]
@barthalion Could you hint me when to expect publication on flathub? Currently returns 404 https://flathub.org/apps/details/dev.rdm.RDM

**Comment by barthalion** [comment]
https://flathub.org/apps/details/dev.rdm.RDM works now. There's a default 3 hours delay but you can log in at https://flathub.org/builds and manually publish your build, just in case.

**Comment by uglide** [comment]
@barthalion thanks for the info

**Pull request closed without merging**